File: packages/core/src/sampling.ts
import { cubehelix } from "./cubehelix";
import type { CubehelixParams, RGB } from "./types";

export function sampleSequential(params: CubehelixParams, n: number): RGB[] {
  if (!Number.isInteger(n) || n < 2) {
    throw new RangeError(`sampleSequential requires n >= 2 integer; got ${n}`);
  }
  const out: RGB[] = new Array(n);
  for (let i = 0; i < n; i++) {
    out[i] = cubehelix(i / (n - 1), params);
  }
  return out;
}

File: packages/python/src/cubehelix_studio/sampling.py
"""Sequential sampling of the cubehelix curve."""

from __future__ import annotations

from .cubehelix import CubehelixParams, cubehelix


def sample_sequential(params: CubehelixParams, n: int) -> list[tuple[float, float, float]]:
    if isinstance(n, bool) or not isinstance(n, int) or n < 2:
        raise ValueError(f"sample_sequential requires n >= 2 integer; got {n!r}")
    return [cubehelix(i / (n - 1), params) for i in range(n)]
